sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -699,35 +699,42 @@ def settle_message_via_receiver_link(
         dead_letter_error_description: Optional[str] = None,
     ) -> None:
         # pylint: disable=protected-access
-        if settle_operation == MESSAGE_COMPLETE:
-            return handler.settle_messages(message._delivery_id, 'accepted')
-        if settle_operation == MESSAGE_ABANDON:
-            return handler.settle_messages(
-                message._delivery_id,
-                'modified',
-                delivery_failed=True,
-                undeliverable_here=False
-            )
-        if settle_operation == MESSAGE_DEAD_LETTER:
-            return handler.settle_messages(
-                message._delivery_id,
-                'rejected',
-                error=AMQPError(
-                    condition=DEADLETTERNAME,
-                    description=dead_letter_error_description,
-                    info={
-                        RECEIVER_LINK_DEAD_LETTER_REASON: dead_letter_reason,
-                        RECEIVER_LINK_DEAD_LETTER_ERROR_DESCRIPTION: dead_letter_error_description,
-                    }
+        if not handler:
+            raise RuntimeError("handler is None")
```

Comment:
Handler is None

Handler is closed?

Reply:
would we hit this at all with the change made below?

Because wouldn't you have an AMQPConnectionError to start with when trying to settle and then go through this function and raise a RunTimeError and then in the SBReceiver retry via mgmt_link, which would close/re-open the handler and then retry settling and fail because it is on a different Link

Reply:
also, I think if we want the error to be caught below, it should be moved inside the `try` block to line 706.

Reply:
For the flow of this error to happen, one has to be stuck in between say a receive_messages call and a send_completion call. As it resumes and a connection disconnect has been issued by the service, everything starts shutting down. So the first time you come here,  handler could be instantiated but pyamqp raises an `AMQPConnectionError` that the Connection has been dropped and then the next time around you come around handler is closed and is None. That would raise the NoneType Attribute error. So the handler check is for the second time around after which completion is tried over managment link ( which opens up handler )